ADR/####-gitops-onboarding-redesign.md

```diff
@@ -0,0 +1,280 @@
+- CI is not under control of the namespace owner
+- Users are forced to use a specific git forge that might be different than the one their apps live in
+
+### Current Pain Points
+1. **Fragmented experience**: Developers must context-switch between UI and GitOps approaches
+2. **Poor experience**: Onboarding / creating new objects requires authoring complex YAML objects by using examples from documentation or other repositories.
+3. **Error-prone process**: Manual YAML authoring can lead to frequent failures
+4. **Monorepo complexity**: Current GitOps monorepo for tenant configuration creates CODEOWNERS management overhead, requires freqeuent rebases and is prone to merge conflicts
+5. **UI improvement limitations**: Despite continuous improvements to the UI, many users still choose GitOps onboarding for its inherent disaster recovery and change management benefits, leaving them unable to benefit from UI enhancements
+
+## Decision
+
+Redesign the onboarding experience to converge UI and GitOps into a single coherent, Git-centric flow through the following architectural changes:
+
+### Alternative Approaches Considered
+
+**Improving the existing UI** was considered but rejected because:
+- Despite continuous UI improvements, many users still choose GitOps onboarding for its inherent disaster recovery and change management benefits
+- UI enhancements don't address the fundamental issue of maintaining two separate onboarding flows
+- Users who prefer GitOps principles cannot benefit from UI improvements, creating a permanent split in the user experience
+- Konflux needs a strong disaster recovery story and having Gitops-based tenant configuration is a significant advantage there.
+
+**Chosen approach** replaces both current flows with a unified Git-centric solution that preserves GitOps benefits while providing the usability improvements users expect.
+
+### Core Shift
+- **Deprecate ability to configure tenant resources via the UI** Users will configure resources directly via Kubernetes or via Gitops
+- **Promote use of Git as the single source of truth** for applications, components, integration test scenarios, releasePlans and RBAC in production environments
+- **Preserve the UI for non-configuration activities** including:
+  - Monitoring and observability: logs, build inspection, viewing metrics, dashboards
+  - Operational actions: triggering builds/tests, starting manual releases, pipeline management
+  - Any other runtime actions that don't involve declarative resource configuration
+- **Commit to single onboarding flow**: UI-based onboarding will be completely deprecated with no fallback option to minimize maintenance overhead and deliver optimal user outcomes
+- **Maintain ease of use and excellent user experience** by providing a VS Code plugin that delivers forms based configuration of validation of tenant resources
+
+### IDE Selection Rationale
+VS Code was chosen as the primary IDE target because:
```

Comment:
While VSCode is popular choice, it still leaves behind many users. I think a CLI is a better option here.